File: haskell/pe003.hs
largestPrimeFactor :: Int -> Int
largestPrimeFactor n = lpf n 3 where
                       lpf num d
                         | num == d         = d
                         | num `rem` d == 0 = lpf (num `div` d) d
                         | otherwise        = lpf num (d+2)
main :: IO ()
main = print $ largestPrimeFactor 600851475143

File: rust/pe003.rs
fn main() {
    let mut num:i64 = 600851475143;
    let mut divisor = 2;
    while num != divisor {
        if num % divisor == 0 {
            num /= divisor;
            divisor-=1;
        }
        divisor+=1;
    }
    println!("{}", divisor);
}